File: trees/app/src/main/java/com/trees/model/ImageViewModelFactory.java
package com.trees.model;


import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.trees.common.helpers.ImageStoreInterface;
import com.trees.common.jni.ImageProcessorInterface;

import java.lang.reflect.InvocationTargetException;

public class ImageViewModelFactory implements ViewModelProvider.Factory {
    private ImageStoreInterface imageStoreInterface;
    private ImageProcessorInterface imageProcessorInterface;


    public ImageViewModelFactory(
            ImageProcessorInterface imageProcessorInterface, ImageStoreInterface imageStoreInterface ) {
        this.imageProcessorInterface = imageProcessorInterface;
        this.imageStoreInterface = imageStoreInterface;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        try {
            return modelClass.getConstructor(
                    ImageProcessorInterface.class, ImageStoreInterface.class).newInstance(
                    imageProcessorInterface, imageStoreInterface);
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        } catch (InstantiationException e) {
            e.printStackTrace();
        } catch (InvocationTargetException e) {
            e.printStackTrace();
        } catch (NoSuchMethodException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: trees/app/src/main/java/com/trees/model/ImageViewModelFactory.kt
package com.trees.model

import android.os.Bundle
import androidx.lifecycle.AbstractSavedStateViewModelFactory
import androidx.lifecycle.SavedStateHandle
import androidx.lifecycle.ViewModel
import androidx.savedstate.SavedStateRegistryOwner
import com.trees.common.helpers.ImageStoreInterface
import com.trees.common.pyi.ImageProcessorInterface

class ImageViewModelFactory(
        private var imageStoreInterface: ImageStoreInterface? = null,
        private var imageProcessorInterface: ImageProcessorInterface? = null,
        owner: SavedStateRegistryOwner,
        defaultArgs: Bundle? = null
) : AbstractSavedStateViewModelFactory(owner, defaultArgs) {
    override fun <T : ViewModel> create(
            key: String,
            modelClass: Class<T>,
            handle: SavedStateHandle
    ): T {
        return ImageViewModel(handle, imageProcessorInterface, imageStoreInterface) as T
    }
}